frontend/src/Editor/Viewer.jsx

```diff
@@ -150,28 +160,41 @@ class Viewer extends React.Component {
                     width: currentLayout === 'desktop' ? '1292px' : `${deviceWindowWidth}px`,
                   }}
                 >
-                  {defaultComponentStateComputed && 
-                    <Container
-                      appDefinition={appDefinition}
-                      appDefinitionChanged={() => false} // function not relevant in viewer
-                      snapToGrid={true}
-                      appLoading={isLoading}
-                      darkMode={this.props.darkMode}
-                      onEvent={(eventName, options) => onEvent(this, eventName, options, 'view')}
-                      mode="view"
-                      scaleValue={scaleValue}
-                      deviceWindowWidth={deviceWindowWidth}
-                      currentLayout={currentLayout}
-                      currentState={this.state.currentState}
-                      onComponentClick={(id, component) => onComponentClick(this, id, component, 'view')}
-                      onComponentOptionChanged={(component, optionName, value) =>
-                        onComponentOptionChanged(this, component, optionName, value)
-                      }
-                      onComponentOptionsChanged={(component, options) =>
-                        onComponentOptionsChanged(this, component, options)
-                      }
-                    />
-                  }
+
+
+                {defaultComponentStateComputed && (
+                  <>
+                    {appLoading ? (
+                      <div className="mx-auto mt-5 w-50 p-5">
+                        <center>
+                          <div className="spinner-border text-azure" role="status"></div>
+                        </center>
+                      </div>
+                    ) : (
+                      <Container
+                        appDefinition={appDefinition}
+                        appDefinitionChanged={() => false} // function not relevant in viewer
+                        snapToGrid={true}
+                        appLoading={isLoading}
```

Comment:
This line is also confusing, should it be changed to `appLoading={appLoading}`? I think this part of the code requires some refactoring.